pkg/controller/manage/manage_controller.go

```diff
@@ -337,3 +343,79 @@ func sendCertRequest(security *kmeshsecurity.SecretManager, pod *corev1.Pod, op
 		security.SendCertRequest(Identity, op)
 	}
 }
+
+func linkXdp(netNsPath string, xdpProgFd int, mode string) error {
+	// Currently only support workload mode
+	if mode != constants.WorkloadMode {
+		return nil
+	}
+
+	if err := netns.WithNetNSPath(netNsPath, func(netns.NetNS) error {
+		// Get all NIC iface in a pod
+		ifaces, err := net.Interfaces()
+		if err != nil {
+			return err
+		}
+
+		// Link XDP prog on every iface, except loopback or not up
+		for _, iface := range ifaces {
+			if iface.Flags&net.FlagLoopback != 0 || iface.Flags&net.FlagUp == 0 {
+				continue
+			}
+			ifLink, err := netlink.LinkByName(iface.Name)
+			if err != nil {
+				return err
+			}
+			// Always let new XDP program replace the old one, to ensure that there is always only one XDP program at the same time
+			if err := netlink.LinkSetXdpFd(ifLink, xdpProgFd); err != nil {
+				return err
+			}
+		}
+		return nil
+	}); err != nil {
+		log.Errorf("Run link xdp in netNsPath %v failed, err: %v", netNsPath, err)
+		return err
+	}
+
+	return nil
+}
+
+func unlinkXdp(netNsPath string, mode string) error {
+	// Currently only support workload mode
+	if mode != constants.WorkloadMode {
+		return nil
+	}
+
+	if err := netns.WithNetNSPath(netNsPath, func(netns.NetNS) error {
+		// Get all NIC iface in a pod
+		ifaces, err := net.Interfaces()
+		if err != nil {
+			return err
+		}
+
+		// Unlink XDP prog on every iface, except loopback or not up
+		for _, iface := range ifaces {
+			if iface.Flags&net.FlagLoopback != 0 || iface.Flags&net.FlagUp == 0 {
+				continue
+			}
+			ifLink, err := netlink.LinkByName(iface.Name)
+			if err != nil {
+				return err
+			}
+			// Detach by using netlink since pin doesn't exist
+			if err := netlink.LinkSetXdpFdWithFlags(ifLink, -1, int(link.XDPGenericMode)); err != nil {
+				return fmt.Errorf("detaching generic-mode XDP program using netlink: %w", err)
+			}
+
+			if err := netlink.LinkSetXdpFdWithFlags(ifLink, -1, int(link.XDPDriverMode)); err != nil {
```

Comment:
why `XDPDriverMode` and `XDPGenericMode` needed

Reply:
reference to cilium's XDP detach, because we do not know which mode(Driver or Generic) the XDP is linked